variables.tf

```diff
@@ -0,0 +1,109 @@
+  }
+}
+
+variable "auto_initialization_using_recovery_crypto_units" {
+  type        = bool
+  description = "Set to true if auto initialization using recovery crypto units is required"
+  default     = false
+}
+
+variable "number_of_crypto_units" {
+  type        = number
+  description = "The number of operational crypto units for your service instance"
+  default     = 2
+  validation {
+    condition     = contains([2, 3], var.number_of_crypto_units)
+    error_message = "Allowed value of number_of_crypto_units is 2 or 3"
+  }
+}
+
+variable "tags" {
+  description = "Optional list of resource tags to apply to resources created"
+  type        = list(string)
+  default     = []
+}
+
+variable "signature_threshold" {
+  type        = number
+  description = "The number of administrator signatures that is required to execute administrative commands"
+  default     = 1
+  validation {
+    condition     = var.signature_threshold >= 1 && var.signature_threshold <= 8
+    error_message = "Allowed values of signature_threshold is between 1 and 8"
+  }
+}
+
+variable "revocation_threshold" {
+  type        = number
+  description = "The number of administrator signatures that is required to remove an administrator after you leave imprint mode"
+  default     = 1
+  validation {
+    condition     = var.revocation_threshold >= 1 && var.revocation_threshold <= 8
+    error_message = "Allowed values of revocation_threshold is between 1 and 8"
+  }
+}
+
+variable "signature_server_url" {
+  type        = string
+  description = "The URL and port number of the signing service. Required if you are using a third-party signing service to provide administrator signature keys"
+  default     = ""
+}
+
+variable "admins" {
+  type = list(object({
+    name = string # max length: 30 chars
+    key  = string # the absolute path and the file name of the signature key file if key files are created using TKE CLI and are not using a third-party signing service
+    # if you are using a signing service, the key name is appended to a URI that will be sent to the signing service
+    token = string # sensitive: the administrator password/token to authorize and access the corresponding signature key file
+  }))
+  default     = []
+  sensitive   = true
+  description = "A list of administrators for the instance crypto units. You can set up to 8 administrators."
+}
+
+variable "number_of_failover_units" {
+  type        = number
+  description = "The number of failover crypto units for your service instance. Default is 0 if not specified and cross-region high availability will not be enabled."
+  default     = 0
+  validation {
+    condition     = contains([0, 2, 3], var.number_of_failover_units)
+    error_message = "Allowed values of failover_units is 0, 2, 3."
+  }
+}
+
+variable "service_endpoints" {
+  type        = string
+  description = "The service_endpoints to access your service instance. Default value is public-and-private if not specified."
```

Comment:
"if not specified" could be interpreted as passing `null` value, and that is not true, so perhaps remove that part from description

Reply:
`nullable = false` in description? In this case we have variable validation which wont allow null value, so it is taken care.

Reply:
ok, got it. Added.